processors/blockprocessor/block_processor.go

```diff
@@ -0,0 +1,367 @@
+	return uint64(proc.ledger.Latest()) + 1
+}
+
+// Preload all resources (account data, account resources, asset/app creators) for the
+// evaluator.
+func prepareEvalResources(l *indexerledger.LedgerForEvaluator, block *bookkeeping.Block) (ledger.EvalForIndexerResources, error) {
+	assetCreators, appCreators, err := prepareCreators(l, block.Payset)
+	if err != nil {
+		return ledger.EvalForIndexerResources{},
+			fmt.Errorf("prepareEvalResources() err: %w", err)
+	}
+
+	res := ledger.EvalForIndexerResources{
+		Accounts:  nil,
+		Resources: nil,
+		Creators:  make(map[ledger.Creatable]ledger.FoundAddress),
+	}
+
+	for index, foundAddress := range assetCreators {
+		creatable := ledger.Creatable{
+			Index: basics.CreatableIndex(index),
+			Type:  basics.AssetCreatable,
+		}
+		res.Creators[creatable] = foundAddress
+	}
+	for index, foundAddress := range appCreators {
+		creatable := ledger.Creatable{
+			Index: basics.CreatableIndex(index),
+			Type:  basics.AppCreatable,
+		}
+		res.Creators[creatable] = foundAddress
+	}
+
+	res.Accounts, res.Resources, err = prepareAccountsResources(l, block.Payset, assetCreators, appCreators)
+	if err != nil {
+		return ledger.EvalForIndexerResources{},
+			fmt.Errorf("prepareEvalResources() err: %w", err)
+	}
+
+	return res, nil
+}
+
+// Preload asset and app creators.
+func prepareCreators(l *indexerledger.LedgerForEvaluator, payset transactions.Payset) (map[basics.AssetIndex]ledger.FoundAddress, map[basics.AppIndex]ledger.FoundAddress, error) {
+	assetsReq, appsReq := accounting.MakePreloadCreatorsRequest(payset)
+
+	assets, err := l.GetAssetCreator(assetsReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareCreators() err: %w", err)
+	}
+	apps, err := l.GetAppCreator(appsReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareCreators() err: %w", err)
+	}
+
+	return assets, apps, nil
+}
+
+// Preload account data and account resources.
+func prepareAccountsResources(l *indexerledger.LedgerForEvaluator, payset transactions.Payset, assetCreators map[basics.AssetIndex]ledger.FoundAddress, appCreators map[basics.AppIndex]ledger.FoundAddress) (map[basics.Address]*ledgercore.AccountData, map[basics.Address]map[ledger.Creatable]ledgercore.AccountResource, error) {
+	addressesReq, resourcesReq :=
+		accounting.MakePreloadAccountsResourcesRequest(payset, assetCreators, appCreators)
+
+	accounts, err := l.LookupWithoutRewards(addressesReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareAccountsResources() err: %w", err)
+	}
+	resources, err := l.LookupResources(resourcesReq)
+	if err != nil {
+		return nil, nil, fmt.Errorf("prepareAccountsResources() err: %w", err)
+	}
+
+	return accounts, resources, nil
+}
+
+// MakeLegacyProcessorHandlerFunction makes a legacy function that emulates original behavior of block processor
+func MakeLegacyProcessorHandlerFunction(proc *BlockProcessor, handler func(block *ledgercore.ValidatedBlock) error) func(cert *rpcs.EncodedBlockCert) error {
```

Comment:
nit: Consider renaming to use `Adapter` instead of `Legacy`. Something like `MakeBlockProcessorHandlerAdapter`